src/renderer/renderer.js

```diff
@@ -36,8 +70,16 @@ export class SDFGRenderer {
         // Toolbar-related fields
         this.menu = null;
         this.toolbar = null;
+        this.panmode_btn = null;
         this.movemode_btn = null;
         this.selectmode_btn = null;
+        this.addmode_btns = null;
```

Comment:
would be really nice if those were part of an `SDFGEditor` that extends `SDFGRenderer` in the vscode repo

Reply:
I agree. I've spent some time checking what could be moved, and there are significant portions of the current base renderer that can be moved to a new `SDFGEditor extends SDFGRenderer`. Since it doesn't add functionality and is a large refactor, I would vote to do this as part of Phase 2's transition to an NPM package. What do you think?

Reply:
ok